### PR TITLE
actions: Use cilium sysdump

### PR DESCRIPTION
**Pull request by michi-covalent**
We are deprecating python cilium-sysdump.zip soon.

Signed-off-by: Michi Mutsuzaki <michi@isovalent.com>

—

(@bmcustodio) This PR also contains a fix for sysdump itself: since sysdump tasks can submit subtasks, it may happen for small values of `--worker-count` that at some point all worker goroutines are stuck at `Submit` calls. This addresses that by setting the actual number of worker goroutines to the minimum necessary to `Submit` and still being able to continue processing work.

### CODE REVIEW

**Comment by bmcustodio** [comment]
Connectivity tests [failed](https://github.com/cilium/cilium-cli/pull/405/checks?check_run_id=3019986156#step:15:2027) for `EKS (ENI)` but the sysdump fix seems to have been [effective](https://github.com/cilium/cilium-cli/pull/405/checks?check_run_id=3019986156#step:15:2117) 🎉

File: sysdump/sysdump.go [hunk 1]
```diff
@@ -639,10 +639,23 @@ func (c *Collector) Run() error {
...
+	// Adjust the worker count to make enough headroom for tasks that submit sub-tasks.
+	// This is necessary because 'Submit' is blocking.
+	wc := 1
+	for _, t := range tasks {
+		if t.CreatesSubtasks && !c.shouldSkipTask(t) {
+			wc++
+		}
+	}
+	// Take the maximum between the specified worker count and the minimum number of workers required.
+	wc = int(math.Max(float64(wc), float64(c.options.WorkerCount)))
```
Comment (tklauser) [inline sysdump/sysdump.go:651]: I think we can avoid the `int -> float64 -> int` conversion and `math` import here:

```suggestion
	if wc < c.options.WorkerCount {
		wc = c.options.WorkerCount
	}
```

Comment (bmcustodio) [reply sysdump/sysdump.go]: Indeed — thanks @tklauser 🚀

**Comment by michi-covalent** [comment]
i can't self-approve, but lgtm

File: internal/cli/cmd/sysdump.go [hunk 2]
```diff
@@ -102,7 +102,7 @@ func newCmdSysdump() *cobra.Command {
 		"Whether to enable quick mode (i.e. skip collection of 'cilium-bugtool' output and logs)")
 	cmd.Flags().IntVar(&sysdumpOptions.WorkerCount,
 		"worker-count", sysdump.DefaultWorkerCount,
-		"The number of workers to use")
+		"The number of workers to use\nNOTE: If a small value is specified, the actual number of workers may be adjusted automatically")
```
Comment (rolinh) [inline internal/cli/cmd/sysdump.go:105]: I'd rephrase this slightly, maybe something like the following:

> NOTE: There is a lower bound requirement on the number of workers for the sysdump operation to be effective. Therefore, for low value, the actual number of workers may be adjusted upwards.

**Review by rolinh — APPROVED** [review]

**Review by nbusseneau — APPROVED** [review]
LGTM.

**Review by ldelossa — APPROVED** [review]
LGTM

**Comment by nbusseneau** [comment]
Unmarking `ready-to-merge` until I have my answer to https://github.com/cilium/cilium-cli/pull/405#pullrequestreview-702439198

**Comment by michi-covalent** [comment]
> Code changes LGTM. Did you test with `pull_request` triggers to try out the changes before merging?

sort of, kind test ran against the pr, and it's passing using `cilium sysdump` command: https://github.com/cilium/cilium-cli/pull/405/checks?check_run_id=3021738148

File: .github/workflows/kind.yaml [hunk 3]
```diff
@@ -98,8 +98,7 @@ jobs:
         run: |
           cilium status
           kubectl get pods --all-namespaces -o wide
-          curl -sLO https://github.com/cilium/cilium-sysdump/releases/latest/download/cilium-sysdump.zip
-          python cilium-sysdump.zip --output cilium-sysdump-out
+          cilium sysdump --debug --output-filename cilium-sysdump-out
```
Comment (nbusseneau) [inline .github/workflows/kind.yaml:101]: Is there any specific reason for having `--debug` here and not in the others? Is this leftover from testing?

**Review by nbusseneau — APPROVED** [review]
> sort of, kind test ran against the pr, and it's passing using cilium sysdump command

I'll blindly trust that it works on all platforms then.

**Comment by tklauser** [comment]
Removing `ready-to-merge` label until https://github.com/cilium/cilium-cli/pull/405#discussion_r666443532 and https://github.com/cilium/cilium-cli/pull/405#discussion_r666882365 are addressed/answered.

File: internal/cli/cmd/sysdump.go (hunk 2 above)
Comment (bmcustodio) [reply internal/cli/cmd/sysdump.go]: Sure, makes sense!

File: .github/workflows/kind.yaml (hunk 3 above)
Comment (bmcustodio) [reply .github/workflows/kind.yaml]: I think it must be a leftover — I've removed it (cc @tklauser)!

**Review by tklauser — APPROVED** [review]

File: .github/workflows/gke.yaml [hunk 4]
```diff
@@ -99,8 +99,7 @@ jobs:
           kubectl logs --timestamps -n kube-system job/cilium-cli
           kubectl exec -n kube-system job/cilium-cli -- cilium status
           kubectl get pods --all-namespaces -o wide
-          curl -sLO https://github.com/cilium/cilium-sysdump/releases/latest/download/cilium-sysdump.zip
-          python cilium-sysdump.zip --output cilium-sysdump-out
+          cilium sysdump --output-filename cilium-sysdump-out
```
Comment (jrajahalme) [inline .github/workflows/gke.yaml:102]: It appears that in this and maybe other workflows `cilium` (cilium-cli) is not installed locally as it is only used in the cloud via job/cilium-cli. This likely means that sysdump also needs to be run as a k8s job?

Now this fails like:
```
/home/runner/work/_temp/ff9b05e1-e1d4-4ae0-b1fb-1f9c7fc6626b.sh: line 4: cilium: command not found
```
and
```
Warning: No files were found with the provided path: cilium-sysdump-out.zip. No artifacts will be uploaded.
```

Comment (michi-covalent) [reply .github/workflows/gke.yaml:102]: @jrajahalme i think this pr from @nbusseneau will address the issue https://github.com/cilium/cilium-cli/pull/423
